R/tm_a_pca.R

```diff
@@ -250,49 +250,60 @@ srv_a_pca <- function(id, datasets, reporter, dat, plot_height, plot_width, ggpl
       response[[i]]$select$multiple <- FALSE
       response[[i]]$select$always_selected <- NULL
       response[[i]]$select$selected <- NULL
-      response[[i]]$select$choices <- datasets$get_varlabels(dataname = response[[i]]$dataname)
-      response[[1]]$select$choices <- setdiff(
-        response[[1]]$select$choices,
-        datasets$get_keys(response[[i]]$dataname)
+      response[[i]]$select$choices <- formatters::var_labels(data[[response[[i]]$dataname]]())
+      response[[i]]$select$choices <- setdiff(
+        response[[i]]$select$choices,
+        unlist(attr(data, "join_keys")$get(response[[i]]$dataname))
       )
     }
 
-    teal.code::init_chunks()
+    dat_selector <- teal.transform::data_extract_srv(id = "dat", data, dat)
```

Comment:
`dat_selector` isn't practically used in this module. The usage of `dat_selector` is wrong:
```
computation <- reactive({
      validate({
        need(
          !is.null(dat_selector()), "Please select data"
        )
      })
```

Above should be changed to `merged$anl_input_r()$columns_source$dat` I guess